bundle/config/mutator/override_compute.go

```diff
@@ -23,10 +23,10 @@ func (m *overrideCompute) Name() string {
 func overrideJobCompute(j *resources.Job, compute string) {
 	for i := range j.Tasks {
 		task := &j.Tasks[i]
-		if task.NewCluster != nil {
+		if task.NewCluster != nil || task.ExistingClusterId != "" || task.ComputeKey != "" || task.JobClusterKey != "" {
 			task.NewCluster = nil
-			task.ExistingClusterId = compute
-		} else if task.ExistingClusterId != "" {
+			task.JobClusterKey = ""
+			task.ComputeKey = ""
 			task.ExistingClusterId = compute
```

Comment:
@lennartkats-db Does the Jobs API allow a job with job clusters defined but not used?

If not, we need to add a pass to remove unused job clusters.

Reply:
Yeah that's fine.